src/crud.rs

```diff
@@ -0,0 +1,56 @@
+use std::{fs::File, io};
+
+struct CPath {
+    path: String,
+}
+
+impl CPath {
+    pub fn new(path: &str) -> Result<Self, io::ErrorKind> {
+        let n = path.len();
+        let i = path.find(".").unwrap_or_default();
+        if i == 0 || i == n - 1 || i != path.rfind(".").unwrap() {
+            Err(io::ErrorKind::InvalidInput)
+        } else {
+            Ok(CPath {
+                path: path.to_string(),
+            })
+        }
+    }
+}
+
+pub fn create_file(file_name: &str) -> Result<File, io::ErrorKind> {
+    let path = CPath::new(file_name)?;
+    match File::create(path.path) {
```

Comment:
`file_name` could be `path` since the input it's not only for filename but can be the entire file path.

Reply:
```suggestion
pub fn create_file(path: &str) -> Result<File, io::ErrorKind> {
    let path = CPath::new(path)?.path;
    match File::create(path) {
```
now the struct `Cpath` deals with `path`. I though this makes more sense.
also `?` operator gives us the inner value of `Ok(v)` So we can access `cpath` in `Cpath` struct right away.